### PR TITLE
 frecent_paths: trim nonexistent directories 

### LINKED ISSUE #6: "dead" directories should be pruned, not `cd`'d too
If a directory which no longer exists is in the store, it should be pruned rather than returned.

### PR DESCRIPTION
**Pull request by euank**
Fixes #6

### CODE REVIEW

**Comment by LinuxMercedes** [comment]
Does this hit the filesystem for every path in the database every time pazi is called? That seems...expensive.

**Comment by euank** [comment]
Every pazi read, so all `z` and `z -i` yes. For adding a new directory it won't I don't think.

For idle reference, fasd's code does a similar thing here I think: https://github.com/clvv/fasd/blob/90b531a5/fasd#L443-L445

Alternatives could be to do a better job of returning iters everywhere and then have a "trimming iter" which only checks directories as the caller accesses them. I haven't built thing correctly for that to be an easy task though.

Another alternative could be periodic gc, though I don't like that.

Do you have any other alternatives to suggest?

File: src/main.rs [hunk 1]
```diff
@@ -239,12 +236,19 @@ alias z='pazi_cd'
...
             // unwrap is safe because of the 'matches.len() == 0' check above.
             print!("{}", matches.last().unwrap().0);
-            process::exit(0);
         }
-    };
-
-    // By default print the frecency
-    for el in frecency.items_with_frecency() {
-        println!("{:.4}\t{}", el.1 * 100f64, el.0);
+    } else if flags.args.len() > 0 {
```
Comment (euank) [inline src/main.rs]: This change has the side effect of making `z` not work with `PAZI_DEBUG=true`. Whoops!

I need to figure out the right way to express this mess in clap I think

Comment (euank) [reply src/main.rs]: Fixed, silly me, forgot I had a positional arg

**Comment by LinuxMercedes** [comment]
Maybe it'd be better to do at the end of `directory_matches`? If we only drop nonexistent stuff we're about to jump to (or are considering jumping to, perhaps), that would at the very least allow some commonly-appearing-and-disappearing paths to stick around, mayhap.

**Comment by euank** [comment]
@LinuxMercedes I started out trying to do that, but ownership reared its ugly head. The abstractions, as built, aren't friendly to even that small change. Doing some cloning might make it work without too much pain, but I think doing it as part of a lazy-iteration-style-thing will end up being better.

If you want to, or think I should, fiddle with that as an intermediate option for now, you/I could, but left to my own devices I think I'd want to punt and go for the proper lazyness stuff a little further down the road.

**Comment by LinuxMercedes** [comment]
That sounds good to me. I'd love to take a stab at implementing laziness in the near future, though!

**Comment by euank** [comment]
I'll file a followup issue.

**Pull request merged**